Sources/Components/Feed/InAppFeedViewModel.swift

```diff
@@ -0,0 +1,368 @@
+                    self.feed.pageInfo.before = self.feed.entries.first?.__cursor
+                    self.feedClientOptions.status = originalStatus
+                    if self.showRefreshIndicator {
+                        self.showRefreshIndicator = false
+                    }
+                }
+            } catch {
+                handleFeedError(error)
+            }
+        }
+        
+        /// Fetches a new page of feed items when the user scrolls to the bottom of the feed.
+        public func fetchNewPageOfFeedItems() async {
+            guard let after = self.feed.pageInfo.after else { return }
+            feedClientOptions.after = after
+            Knock.shared.feedManager?.getUserFeedContent(options: feedClientOptions) { result in
+                DispatchQueue.main.async {
+                    switch result {
+                    case .success(let feed):
+                        self.mergeFeedsForNewPageOfFeed(feed: feed)
+                    case .failure(let error):
+                        self.handleFeedError(error)
+                    }
+                }
+            }
+        }
+        
+        /// Determines whether there is another page of content to fetch when paginaating the feedItem list
+        public func isMoreContentAvailable() -> Bool {
+            return feed.pageInfo.after != nil
+        }
+        
+        // MARK: Message Update Methods
+        
+        /// Archives a specific feed item.
+        public func archiveItem(_ item: Knock.FeedItem) async {
+            do {
+                let message = try await Knock.shared.updateMessageStatus(messageId: item.id, status: .archived)
+                
+                if let index = self.feed.entries.firstIndex(where: { $0.id == message.id }) {
+                    DispatchQueue.main.async { [weak self] in
+                        self?.feed.entries[index].archived_at = Date()
+                        
+                        // remove local message if update was successful and the archived filter is not set to `exclude`
+                        if self?.shouldHideArchived ?? true {
+                            self?.feed.entries.remove(at: index)
+                        }
+                    }
+                    await fetchNewMetaData()
+                }
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in archiveItem: \(error.localizedDescription)")
+            }
+        }
+        
+        /// Archives all items within the specified scope.
+        public func archiveAll(scope: Knock.FeedItemScope) async {
+            let feedOptions = Knock.FeedClientOptions(status: scope, tenant: currentTenantId, has_tenant: feedClientOptions.has_tenant, archived: nil)
+            do {
+                _ = try await Knock.shared.feedManager?.makeBulkStatusUpdate(type: .archived, options: feedOptions)
+                optimisticallyBulkUpdateStatus(status: .archived)
```

Comment:
This optimisticallyBulkUpdate call needs to take into account the `scope` as well.

- If `read`: only archive read items in the feed entries, set `unread_count` to 0 and decrement `total_count` by `unread_count` 
- If `all`: archive everything and reset total count

For both, should ONLY be performing these updates when `shouldOptimisticallyArchive`